flight/protoMap.py

```diff
@@ -0,0 +1,200 @@
+#!/user/bin/python
+
+import math, time
+import numpy as np
+from typing import Tuple, List
+
+# ------ GLOBAL VARIABLES ------ #
+# Units: mm
+SENSOR_W: float = 4.8
+SENSOR_H: float = 3.6
+SENSOR_D: float = 6.0
+
+# Units: percent (0.0 -> 1.0)
+OVERLAP: float = 0.1
+
+# Radius of Earth
+# Units: ft
+RADIUS: float = 20902230.971129
+
+# Bearings for different directions
+SOUTH: float = math.pi
+EAST: float = math.pi / 2
+NW: float = ((7 * math.pi) / 4)
+
+# ------ GLOBAL VARIABLES ------ #
+
+def map(info: Tuple[Tuple[float, float], float], 
+altitude: int = 750, 
+focal_length: float = 4.9,
+) -> List[Tuple[float, float]]:
+    """
+    Calculates the map area, cuts map area into sections, 
+    then creates flight path for drone through sections.
+
+    Parameters
+    ----------
+    info: Tuple[Tuple[float, float], float]
+        The center location in latitude, longitude of the map, and then height of the map
+    altitude: int
+    focal_length: float
+
+    Returns
+    -------
+    path: List[Tuple[float, float]]
+        The path that the drone will take through the map
+
+    Notes
+    -----
+    fov is the Field of View of the image area
+    fovH is the horizontal field of view distance
+    fovV is the vertical field of view distance
+    """
+
+    # Takes in tuple(start location, map height), altitude, and focal_length]
+    # Assumes altitude and focal_length doesn't change
+    # Returns flight path of drone to snap images of map
+
+    # Split info into center of map location and map height
+    center_map: Tuple[float, float] = info[0]
+    map_h: float = info[1]
+
+    # Calculate the width and height of camera image
+    cam_w, cam_h = image_area(altitude, focal_length)
+    #print("Camera Width:", cam_w)
+    #print("Camera Height:", cam_h)
+
+    # Determine map width using MAP_H and aspect ratio 16:9
+    # Units: ft
+    map_w: float = (16 * map_h) / 9
+
+    #print("Map Width:", map_w)
+    #print("Map Height:", map_h)
+
+    # Find center of each piece consider OVERLAP
+    ft_waypoints: List[Tuple[float, float]] = find_centers(map_w, map_h, cam_w, cam_h, OVERLAP)[0]
+    col = find_centers(map_w, map_h, cam_w, cam_h, OVERLAP)[1]
+    #print("Num of waypoints:", len(ft_waypoints))
+
+    # Convert centers to lat, long coordinates
+    coord: List[Tuple[float, float]] = coordinates(ft_waypoints, info[0], map_w, map_h)
+    #print("Coord:", len(coord))
+
+    # Flight path algorithm
+    path = flight_path(coord, col)
+
+    return path
+
+def image_area(altitude: int, focal_length: float) -> Tuple[float, float]:
```

Comment:
This might be some weirdo stuff from your computer, but this function doesn't have a docstring in this commit.